### ISSUE TITLE
sync()'s recycling fence is vacuous when nothing stages, and the training loop already hits that

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#87]
`sync()` recycles every transient buffer back to the pool at the end of the window, and the comment that justifies it reads:

```ts
// The mapAsync completions above prove all submitted work finished, so
// every transient buffer is idle and safe to recycle.
for (const tr of this.transients) this.pool.release(tr.size, tr.buf);
```

Those completions are the `await`s over `stagings`. When `stagings` is empty there are none, so the sentence proves nothing about the submitted work: the loop releases buffers on the strength of a fence that did not happen. Whether it is nonetheless safe rests on queue ordering, which is a different argument and is not the one written down.

It is not hypothetical. `src/backend/train-gpu.ts:226` reaches it every step with clipping off:

```ts
await gpu.sync(normTensors); // flush the optimizer; read norms iff clipping
```

with both param groups kept on device, so nothing stages and `normTensors` is empty on that path.

Found while reviewing #86, which does not add a new instance: the parity probe always stages its loss scalar. Pre-existing and separate, hence its own issue.

Two ways out, and picking between them is the work:

- If the ordering argument holds, write it down instead. WebGPU submissions on one queue complete in order, so a `submit()` that follows every recorded pass means the pool cannot hand a buffer to a later pass that overtakes an earlier one. That is a claim about the spec, and it should cite the spec rather than a fence.
- If it does not hold, the empty-`stagings` path needs a real fence, `onSubmittedWorkDone()` being the obvious one, and then the cost of adding one per optimizer flush is worth measuring before taking it.

Either way the reviewable artifact is small: a comment that says what actually makes the recycle safe, and a gate if the answer turns out to be the second one.
